test/unit_test/static/unit_constants.cpp

```diff
@@ -0,0 +1,83 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2018 Mateusz Pusz
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#include "units/physical/dimensions/length.h"
+#include "units/physical/dimensions/speed.h"
+#include "units/physical/si/base/length.h"
+#include "units/physical/si/base/mass.h"
+#include "units/physical/si/base/time.h"
+#include "units/physical/si/derived/area.h"
+#include "units/physical/si/derived/speed.h"
+#include <cassert>
+#include <cstdint>
+#include <type_traits>
+
+using namespace units::physical;
+using namespace si;
+using namespace unit_constants;
+
+static_assert(2 * m == 2_q_m);
+static_assert(2 * s == 2_q_s);
+#if !defined(_MSC_VER)
```

Comment:
Could you please use `COMP_MSVC`?

BTW, if you feel we can do better here, we can think about an alternative/better way to refer to compilers in a unified way...

Reply:
For starters, all kind of global identifiers should be "namespaced". In this case, they should be prefixed with `UNITS_` or `MP_UNITS_` (just like the CMake options).